__tests__/app.test.js

```diff
@@ -197,7 +197,38 @@ describe("GET /api/articles", () => {
             .get("/api/articles?order=invalid-order")
             .expect(400)
             .then(({ body: { msg } }) => {
-                expect(msg).toBe("Bad Request")
+                expect(msg).toBe("Bad Request: Invalid Order Query")
+            })
+    })
+    test("200: returns array of articles filtered by the topic specified in the query", () => {
+        return request(app)
+            .get("/api/articles?topic=mitch")
+            .expect(200)
+            .then(({ body: { articles } }) => {
+                expect(articles.length).toBe(12)
+                expect(articles).toBeSortedBy("created_at", {
+                    descending: true,
+                })
+                articles.forEach((article) => {
+                    expect(article).not.toContainKey("body")
+                    expect(article).toMatchObject({
+                        title: expect.any(String),
+                        topic: "mitch",
+                        author: expect.any(String),
+                        created_at: expect.any(String),
+                        votes: expect.any(Number),
+                        article_img_url: expect.any(String),
+                        comment_count: expect.any(Number),
+                    })
+                })
+            })
+    })
+    test("400: sends an appropriate status and error message when given an invalid topic query", () => {
+        return request(app)
+            .get("/api/articles?topic=invalid-topic")
+            .expect(400)
+            .then(({ body: { msg } }) => {
+                expect(msg).toBe("Bad Request: Invalid Topic Query")
```

Comment:
I would expect to see a 200 test for a valid topic with no articles. Such as paper.